init-template/SourceSolution/Blahem.E2eTests/Shared/Endpoints/BlaitemEndpoints.cs

```diff
@@ -12,45 +14,67 @@ internal static class BlaitemHttpClientExtensions
     public static DeleteBlaitemEndpoint DeleteBlaitem(this HttpClient httpClient) => new(httpClient);
 }
 
-internal class CreateBlaitemEndpoint : PostApiEndpointWithDto<CreateBlaitemDto, int>
+internal class CreateBlaitemEndpoint : ApiEndpointBase<CreateBlaitemDto, int>
 {
     internal CreateBlaitemEndpoint(HttpClient httpClient) :
         base(httpClient)
     {
     }
 
-    internal override string BuildPath(CreateBlaitemDto dto) => $"api/blaitems";
+    public override async Task<HttpResponseMessage> Call(CreateBlaitemDto dto)
+    {
+        var path = "api/blaitems";
+
+        return await HttpClient.PostAsJsonAsync(path, dto);
+    }
 }
 
-internal class GetBlaitemByIdEndpoint : GetApiEndpointWithDto<int, BlaitemDto>
+internal class GetBlaitemByIdEndpoint : ApiEndpointBase<int, BlaitemDto>
 {
     internal GetBlaitemByIdEndpoint(HttpClient httpClient) :
         base(httpClient)
     {
     }
 
-    internal override string BuildPath(int blaitemId) => $"api/blaitems/{blaitemId}";
+    public override async Task<HttpResponseMessage> Call(int blaitemId)
+    {
+        var path = $"api/blaitems/{blaitemId}";
+
+        return await HttpClient.GetAsync(path);
+    }
 }
 
-internal class ListBlaitemsEndpoint : GetApiEndpoint<List<BlaitemDto>>
+internal class ListBlaitemsEndpoint : ApiEndpointBase<PaginatedListQuery, PaginatedListResponse<BlaitemDto>>
 {
-    internal ListBlaitemsEndpoint(HttpClient httpClient) :
-        base(httpClient, "api/blaitems")
+    public ListBlaitemsEndpoint(HttpClient httpClient) :
+        base(httpClient)
     {
     }
+
+    public override async Task<HttpResponseMessage> Call(PaginatedListQuery dto)
+    {
+        var path = $"api/blaitems?{dto.ToQueryString()}";
+
+        return await HttpClient.GetAsync(path);
+    }
 }
 
-internal class UpdateBlaitemEndpoint : PutApiEndpointWithDto<UpdateBlaitemDto, int>
+internal class UpdateBlaitemEndpoint : ApiEndpointBase<UpdateBlaitemDto, int>
 {
     internal UpdateBlaitemEndpoint(HttpClient httpClient) :
         base(httpClient)
     {
     }
 
-    internal override string BuildPath(UpdateBlaitemDto dto) => $"api/blaitems/{dto.Id}";
+    public override async Task<HttpResponseMessage> Call(UpdateBlaitemDto dto)
+    {
+        var path = $"api/blaitems/{dto.Id}";
+
+        return await HttpClient.PutAsJsonAsync(path, dto);
+    }
 }
 
-internal class DeleteBlaitemEndpoint : DeleteApiEndpoint
+internal class DeleteBlaitemEndpoint : ApiEndpointBase<int, int>
```

Comment:
Same here, this should inherit from ApiEndpointBaseWithoutResponse as Delete endpoints have no response content.